controllers/conditions.go

```diff
@@ -0,0 +1,133 @@
+	for i := range conditions {
+		if conditions[i].Type == condType {
+			return &conditions[i]
+		}
+	}
+	return nil
+}
+
+func setClusterInstallCondition(conditions *[]hivev1.ClusterInstallCondition, newCondition hivev1.ClusterInstallCondition) {
+	if conditions == nil {
+		return
+	}
+	now := metav1.NewTime(time.Now())
+	existingCondition := findCondition(*conditions, newCondition.Type)
+	if existingCondition == nil {
+		if newCondition.LastTransitionTime.IsZero() {
+			newCondition.LastTransitionTime = now
+		}
+		if newCondition.LastProbeTime.IsZero() {
+			newCondition.LastProbeTime = now
+		}
+		*conditions = append(*conditions, newCondition)
+		return
+	}
+
+	if existingCondition.Status != newCondition.Status {
+		existingCondition.Status = newCondition.Status
+		if !newCondition.LastTransitionTime.IsZero() {
+			existingCondition.LastTransitionTime = newCondition.LastTransitionTime
+		} else {
+			existingCondition.LastTransitionTime = now
+		}
+	}
+	if !newCondition.LastProbeTime.IsZero() {
+		existingCondition.LastProbeTime = newCondition.LastProbeTime
+	} else {
+		existingCondition.LastProbeTime = now
+	}
+
+	existingCondition.Reason = newCondition.Reason
+	existingCondition.Message = newCondition.Message
+}
+
+func (r *ImageClusterInstallReconciler) initializeConditions(ctx context.Context, ici *v1alpha1.ImageClusterInstall) error {
+	initialTypeStatus := map[hivev1.ClusterInstallConditionType]corev1.ConditionStatus{
+		hivev1.ClusterInstallRequirementsMet: corev1.ConditionUnknown,
+		hivev1.ClusterInstallCompleted:       corev1.ConditionUnknown,
+		hivev1.ClusterInstallFailed:          corev1.ConditionUnknown,
+		hivev1.ClusterInstallStopped:         corev1.ConditionFalse,
+	}
+
+	patch := client.MergeFrom(ici.DeepCopy())
+	for condType, status := range initialTypeStatus {
+		if findCondition(ici.Status.Conditions, condType) == nil {
+			setClusterInstallCondition(&ici.Status.Conditions, hivev1.ClusterInstallCondition{
+				Type:   condType,
+				Status: status,
+			})
+		}
+	}
+
+	return r.Status().Patch(ctx, ici, patch)
+}
+
+func (r *ImageClusterInstallReconciler) setImageReadyCondition(ctx context.Context, ici *v1alpha1.ImageClusterInstall, err error) error {
+	cond := hivev1.ClusterInstallCondition{
+		Type:    hivev1.ClusterInstallRequirementsMet,
+		Status:  corev1.ConditionTrue,
+		Reason:  v1alpha1.ImageReadyReason,
+		Message: v1alpha1.ImageReadyMessage,
+	}
+
+	if err != nil {
+		cond.Status = corev1.ConditionFalse
+		cond.Reason = v1alpha1.ImageNotReadyReason
+		cond.Message = err.Error()
+	}
+
+	patch := client.MergeFrom(ici.DeepCopy())
+	setClusterInstallCondition(&ici.Status.Conditions, cond)
+	return r.Status().Patch(ctx, ici, patch)
+}
+
+func (r *ImageClusterInstallReconciler) setHostConfiguredCondition(ctx context.Context, ici *v1alpha1.ImageClusterInstall, err error) error {
```

Comment:
Where do you call this?

Reply:
The function already existed it just set a different condition before.

Currently it's called here https://github.com/openshift/image-based-install-operator/blob/6f767d683a7091a4831fe27b8b034a79d4f54e8e/controllers/imageclusterinstall_controller.go#L167-L172

Reply:
I debated changing the name to match the new condition type, but these functions are also setting the reason and message to the IBIO-specific ones so I just left it since we'd want it called at the same time.